### DISCUSSION TITLE
How to Make Filenames Take More Space in LuaLine

### DISCUSSION THREAD

**Discussion by StevenXL** [LazyVim/LazyVim#2084] [Q&A] [answered]
Hi folks, 

My lualine status looks something like this:

![image](https://github.com/LazyVim/LazyVim/assets/7391433/8410c1a4-a486-48de-81a9-d213f99044fd)

I would like to replace the `...` with more information on the filename. 

I tried adding a `lualine.lua` file to my `lua/plugins` folder with this:

```lua
return {
  {
    "nvim-lualine/lualine.nvim",
    event = "VeryLazy",
    opts = function(_, opts)
      table.insert(opts.sections.lualine_c, {shorting_target = 0})
    end,
  },
}
```

But that doesn't seem to have any affect. What am I doing wrong?

**Comment by dpetka2001** [comment, marked as answer]
The following worked for me
```lua
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      table.remove(opts.sections.lualine_c)
      table.insert(opts.sections.lualine_c, {
        "filename",
        path = 3,
      })
    end,
  },
```
Otherwise you would have to change the [pretty_path](https://github.com/LazyVim/LazyVim/blob/68ff818a5bb7549f90b05e412b76fe448f605ffb/lua/lazyvim/util/lualine.lua#L66) function yourself and use your customized version I would guess.

**Reply by StevenXL** [reply]
Thanks for the answer. I did something a bit different but based totally on your answer:

```lua
local icons = require("lazyvim.config").icons

return {
  "nvim-lualine/lualine.nvim",
  opts = function(_, opts)
    table.remove(opts.sections.lualine_c)

    table.insert(opts.sections.lualine_c, {
      "filename",
      path = 3,
    })

    table.insert(opts.sections.lualine_c, {
      "diagnostics",
      symbols = {
        error = icons.diagnostics.Error,
        warn = icons.diagnostics.Warn,
        info = icons.diagnostics.Info,
        hint = icons.diagnostics.Hint,
      },
    })
  end,
}
```

**Comment by linkvt** [comment]
A bit late but I found a different answer that's possible due to more recent changes how the `pretty_path` function works.

We can now pass opts to the `pretty_path` function to determine how many path parts should be displayed before being abbreviated to `...`, see: https://github.com/LazyVim/LazyVim/blob/90f84e7e755f72bb99dc8a9dedbfef05e58eb09d/lua/lazyvim/util/lualine.lua#L124 

I added a `lualine.lua` to my plugins folder and overwrote the 4th argument, where `pretty_path` is called in section C without any params by default:
```lua
return {
  "nvim-lualine/lualine.nvim",
  optional = true,
  opts = function(_, opts)
    opts.sections.lualine_c[4] = {
      LazyVim.lualine.pretty_path({
        length = 0, -- 0 parts means nothing is abbreviated, see https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/util/lualine.lua#L124
      }),
    }
  end,
}
```

Accessing the index explicitly is of course not really stable so you might need to adjust it in the future if the order changes by comparing the [current](https://github.com/LazyVim/LazyVim/blob/90f84e7e755f72bb99dc8a9dedbfef05e58eb09d/lua/lazyvim/plugins/ui.lua#L109) config to the [new one](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/ui.lua).
